pkg/udf/forward/forward.go

```diff
@@ -249,50 +239,37 @@ func (isdf *InterStepDataForward) forwardAChunk(ctx context.Context) {
 	// we fetch the watermark for the partition from which we read the message.
 	processorWM := isdf.wmFetcher.ComputeWatermark(readMessages[0].ReadOffset, isdf.fromBufferPartition.GetPartitionIdx())
 
+	// assign watermark to data messages
+	for _, msg := range dataMessages {
+		msg.Watermark = time.Time(processorWM)
+	}
+
+	var udfResults []isb.ReadWriteMessagePair
 	var writeOffsets map[string][][]isb.Offset
-	if !isdf.opts.enableMapUdfStream {
+	// Check if map streaming mode is enabled
+	if isdf.opts.enableMapUdfStream {
```

Comment:
this check will go away, right?

Reply:
Not right now, in this PR we would not unify all the map functionalities. Would be better to do in a follow up PR as a separate task